### PR TITLE
Fix for 1170

### PR DESCRIPTION
**Pull request by sbrossie**

### CODE REVIEW

**Comment by sbrossie** [comment]
@pierre for review

File: beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithBCDUpdate.java [hunk 1]
```diff
@@ -862,4 +864,32 @@ public void testBCDChangeForConsumableInArrearPlan() throws Exception {
...
+    public void testWithOverrideFixedPriceAndNewBCD() throws Exception {
+
+        final DateTime initialDate = new DateTime(2016, 4, 1, 0, 13, 42, 0, testTimeZone);
+        clock.setTime(initialDate);
+
+        final Account account = createAccountWithNonOsgiPaymentMethod(getAccountData(0));
+        assertNotNull(account);
+
+        // Price override of $10 for fixed and $0 for recurring
+        // Override BCD as well
```
Comment (pierre) [inline beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithBCDUpdate.java]: The account BCD isn't set here though, would it be worth adding a second test where the account has a non zero BCD?

Comment (sbrossie) [reply beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithBCDUpdate.java]: I certainly could, but that seems orthogonal to the issue? The issue is simply that that if there is BCD event in the stream on the same date, we discard fixed price from previous event, and all this is independent of whether account BCD was set or not

Comment (pierre) [reply beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithBCDUpdate.java]: Fair enough 👍
